### ISSUE TITLE
Unwanted "supports" inside the walls

### ISSUE THREAD

**Issue by robustini** [supermerill/SuperSlicer#426]
I don't find the option to remove these supports (in the firt image) inside the perimeters which are generated only if the infill is present.
If I put infill at 0% this addition disappears (second image), yet I hink there is an option in the "Infill" section to not generate them, or is missing?

SuperSlicer V2.2.53

![immagine](https://user-images.githubusercontent.com/3864712/91299539-11593a80-e7a2-11ea-9114-8aeb20a72fb1.png)

![immagine](https://user-images.githubusercontent.com/3864712/91299671-4c5b6e00-e7a2-11ea-886f-392c4765559e.png)

[FUSO1.zip](https://github.com/supermerill/SuperSlicer/files/5129573/FUSO1.zip)

**Comment by neophyl** [comment]
You may want to read this issue in the main Prusa Slicer repository https://github.com/prusa3d/PrusaSlicer/issues/3744

**Comment by Pavulon87** [comment]
> You may want to read this issue in the main Prusa Slicer repository [prusa3d#3744](https://github.com/prusa3d/PrusaSlicer/issues/3744)
> or this one here #253 (took me awhile to find it)

I just lost one hour looking for ensure_vertical_shell_thickness in code to find this post :D
Layer modifier is great work-a-round.

**Comment by robustini** [comment]
I tried the workaround but it produces the same result here.

![immagine](https://user-images.githubusercontent.com/3864712/91310167-cfd08b80-e7b1-11ea-9832-5b51527983b1.png)

**Comment by Pavulon87** [comment]
@robustini, are you sure that "Ensure vertical shell thickness" is turned off?

**Comment by supermerill** [comment]
If ensure_vertical_shell_thikness is deactivated, a little codepath (printobject line 3200) change sparse infill to solid infill if the gap is < 3 * solid extrusion width. This is deactivated if fill_density is 0.

that's the problem here

**Comment by supermerill** [comment]
For now, if you enable "ensure vertiacl thickness", you have much less solid infill. But no zero.

**Comment by Pavulon87** [comment]
> note that there is this comment before this code block:
> //FIXME Vojtech: Disable this and you will be sorry.
> 
> edit: disabling this doesn't seems that problematic.
Maybe it will be a worse solution for some models, but I believe that commenting on it gives a better result.
Much less unnecessary printing after commenting. And without visible problems when the option is activated.

**Comment by robustini** [comment]
> 
> 
> @robustini, are you sure that "Ensure vertical shell thickness" is turned off?

Open my project in the zip file and verify, is disabled but the problem is still there.

**Comment by haleypearse** [comment]
Yes I have the same thought. For two years I've checked every release to see if anyone's bothered to fix this absurd bug. You're right it seems like it would be simple. I believe is written in Perl. Although I'm basically a layperson I'm tempted to have a try.

**Comment by RolfLindbom** [comment]
![Skärmbild 2021-03-27 124734](https://user-images.githubusercontent.com/44773506/112719960-0911db00-8efc-11eb-982a-87e7d1d84e03.png)
![Skärmbild 2021-03-27 130249](https://user-images.githubusercontent.com/44773506/112720195-327f3680-8efd-11eb-9e64-998c88504b48.png)

After some very layman-ish fiddling (I'm an indie Unity gameplay coder...) I found that disabling line 2505 in PrintObject.cpp (PrusaSlicer v2.4 alpha) seems to produce the desired results when "Ensure vertical shell thickness" is deselected.
